homeassistant/components/deconz.py

```diff
@@ -0,0 +1,198 @@
+"""
+Support for deCONZ support.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/deconz/
+"""
+
+import asyncio
+import os
+import logging
+import voluptuous as vol
+
+from homeassistant.config import load_yaml_config_file
+from homeassistant.const import (
+    CONF_API_KEY, CONF_EVENT, CONF_HOST, CONF_ID, CONF_PASSWORD, CONF_PORT,
+    CONF_USERNAME, EVENT_HOMEASSISTANT_STOP)
+from homeassistant.core import callback, EventOrigin
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers import discovery
+from homeassistant.util import slugify
+from homeassistant.util.json import load_json, save_json
+
+REQUIREMENTS = ['pydeconz==20']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'deconz'
+
+DECONZ_DATA = 'deconz_data'
+CONFIG_FILE = 'deconz.conf'
+
+CONF_SWITCH_AS_EVENT = 'switch_as_event'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_HOST): cv.string,
+        vol.Optional(CONF_API_KEY): cv.string,
+        vol.Optional(CONF_PORT, default=80): cv.port,
+        vol.Optional(CONF_SWITCH_AS_EVENT, default=True): cv.boolean,
+        vol.Optional(CONF_USERNAME, default='delight'): cv.string,
+        vol.Optional(CONF_PASSWORD, default='delight'): cv.string,
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+ATTR_FW_VERSION = 'firmware_version'
+ATTR_MANUFACTURER = 'manufacturer'
+ATTR_MODEL_ID = 'model_number'
+ATTR_UNIQUE_ID = 'uniqueid'
+
+
+@asyncio.coroutine
+def async_setup(hass, config):
+    """Setup services for Deconz."""
+    deconz_config = config[DOMAIN]
+    config_file = load_json(hass.config.path(CONFIG_FILE))
+    descriptions = load_yaml_config_file(
+        os.path.join(os.path.dirname(__file__), 'services.yaml'))[DOMAIN]
+
+    @callback
+    def _shutdown(call):  # pylint: disable=unused-argument
+        """Stop the connections to Deconz on shutdown."""
+        if DECONZ_DATA in hass.data:
```

Comment:
Instead of this if check, only register this callback when `hass.data[DECONZ_DATA]` has been set

Reply:
Fixed